src/anomalib/trainer/utils/normalizer/base.py

```diff
@@ -0,0 +1,48 @@
+"""Base normalizer."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABC
+from typing import Type
+
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from torchmetrics import Metric
+
+import anomalib.trainer as trainer
+from anomalib.models import AnomalyModule
+
+
+class BaseNormalizer(ABC):
+    """Base class for normalizers.
+
+    Args:
+        metric_class (Metric): Metric class to be used for normalization.
+        trainer (trainer.AnomalibTrainer): Trainer object.
+    """
+
+    def __init__(self, metric_class: Type[Metric], trainer: "trainer.AnomalibTrainer"):
+        self.metric_class = metric_class
+        self.trainer = trainer
+
+    @property
+    def anomaly_module(self) -> AnomalyModule:
+        """Returns the anomaly module."""
+        if not hasattr(self.trainer, "lightning_module"):
+            raise ValueError("Lightning module is not available yet in trainer.")
+        return self.trainer.lightning_module
+
+    @property
+    def metric(self) -> Metric:
+        """Assigns metrics if not available."""
+        # get metric from anomaly_module
+        if not hasattr(self.trainer.lightning_module, "normalization_metrics"):
+            self.trainer.lightning_module.normalization_metrics = self.metric_class()
+        # Every time metric is called, it is moved to the cpu
+        return self.trainer.lightning_module.normalization_metrics.cpu()
```

Comment:
I think this is somethig might have to address later. We move the metrics and outputs to cpu at a bunch of places. We have this `PostProcessor._outputs_to_cpu(outputs)` in post-processor. I guess the original idea was to avoid oom issues by moving metrics computation to the cpu. But in this case we can look into adding a new key to the config to allow gpu computation.